app/controllers/admin/api/services/proxies_controller.rb

```diff
@@ -63,7 +63,7 @@ def show
   ##~ op.parameters.add name: "jwt_claim_with_client_id_type", description: "JWT Claim With ClientId Type. Either `plain` or `liquid`", dataType: "string", paramType: "query", required: false
   #
   def update
-    if proxy.update_attributes(proxy_params)
+    if proxy.update_attributes(proxy_params) && !apiap?
```

Comment:
@guicassolato will correct me if I'm wrong:
The idea is simply that customer using APIAP will have to deploy manually, that's all. I guess your concern is that this is a quick-workaround rather than a solution. To me, it makes sense to merge this solution temporarily because I understand this is bug in production... and in the meantime we'd work on a more robust one.

Reply:
@hallelujah the goal of the issue is not to automatically deploy when the proxy is updated, yes, but this is just making for the API side what we are already doing for the UI side :smile:

Reply:
Ah one other strange thing that we get rid off :) 👍 
Though we need to put it in the release note that this is a breaking change.

Also we need to think about what to do next when we clean up this mess and everybody will be under apiap

Reply:
Still that means that there is no way to create a proxy config as this controller does not have any `create` method

https://github.com/3scale/porta/blob/ecd417bbcbbf96c83d96e99f61902d6f143f3c8b/app/controllers/admin/api/services/proxy/configs_controller.rb

Reply:
So lets not merge it for 2.7 because this is a breaking change without any deprecation notice.

Customers should not carry our mistakes, so steps to achieve it:

- Deprecate the behaviour (creation of proxy onfig on proxy update) in 2.7
- Add another entry point `#create` in proxy configs or a deploy method in `proxies` controller
- remove the behaviour in 2.8

Reply:
We could also fix the bug instead of changing the behavior. I understand this is a quick solution but it would also be possible to keep the logic as it is *but* preventing the creation of a proxy config when there are no backends.

@guicassolato @hallelujah

Reply:
>Still that means that there is no way to create a proxy config as this controller does not have any create method

I'm not sure I follow, why would we need a create method? Proxy configs are created automatically when deploying and deploying is still possible, right? The only change is that users will have to do it manually.

Reply:
This is all explained in this issue https://issues.jboss.org/browse/THREESCALE-3758 @josemigallas

Reply:
@hallelujah, sorry for taking so long to get back to this. I think this PR is fine and we should move on with it.

This line of code that triggered the discussion means that, by using the **Proxy Update** endpoint, APIAP accounts will be able to save the proxy but this won't deploy the new config to staging anymore. In order to "promote" the new config to staging we already have another endpoint, which is the **Proxy Config Promote**. (Yes, it should be called **Proxy Config Create**!) It accepts an "environment" parameter, which, by passing "sandbox", promotes the latest proxy configs to staging, no?

For non-APIAP accounts, the PR keeps the same behavior.